### PR TITLE
Feature/commit more changes

### PR DESCRIPTION
**Pull request by Arthur-MONNET**

### CODE REVIEW

**Comment by Arthur-MONNET** [comment]
ready for review

File: index.js [hunk 1]
```diff
@@ -170,6 +172,33 @@ const filterLockFiles = (diff) => {
...
+function parseDiffByFile(diff) {
+  const files = [];
+  const lines = diff.split('\n');
+  let currentFile = null;
+  let currentDiff = [];
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i];
+    const diffGitMatch = line.match(/^diff --git a\/(.+?) b\/(.+)$/);
+    if (diffGitMatch) {
+      // Début d'un nouveau fichier
```
Comment (insulineru) [inline index.js:184]: Change language or remove please

File: index.js [hunk 2]
```diff
@@ -170,6 +172,33 @@ const filterLockFiles = (diff) => {
...
+      if (currentFile) {
+        files.push({ filename: currentFile, diff: currentDiff.join('\n') });
+      }
+      currentFile = diffGitMatch[2];
+      currentDiff = [line];
+    } else if (currentFile) {
+      currentDiff.push(line);
+    }
+  }
+  // Ajouter le dernier fichier
```
Comment (insulineru) [inline index.js:194]: Change language or remove please

File: index.js [hunk 3]
```diff
@@ -196,9 +225,97 @@ async function generateAICommit() {
...
 
-  args.list
-    ? await generateListCommits(diff)
-    : await generateSingleCommit(diff);
+  const prompt = getPromptForSingleCommit(diff);
+
+  const numTokens = encode(prompt).length;
+
+  if (numTokens > MAX_TOKENS) {
+    // Diviser le diff par fichier et générer des résumés
```
Comment (insulineru) [inline index.js:233]: Change language or remove please

File: index.js [hunk 4]
```diff
@@ -196,9 +225,97 @@ async function generateAICommit() {
...
+
+    if (!answer.continue) {
+      console.log("Commit aborted by user 🙅‍♂️");
+      process.exit(1);
+    }
+
+    makeCommit(finalCommitMessage);
+
+  } else {
+    // Procéder comme d'habitude si le diff n'est pas trop grand
```
Comment (insulineru) [inline index.js:314]: Change language or remove please

File: package.json [hunk 5]
```diff
@@ -31,9 +31,9 @@
   },
   "homepage": "https://github.com/insulineru/ai-commit#readme",
   "dependencies": {
-    "chatgpt": "^5.0.0",
-    "dotenv": "^16.0.3",
+    "chatgpt": "^5.2.5",
+    "dotenv": "^16.4.5",
     "gpt-3-encoder": "^1.1.4",
-    "inquirer": "^9.1.4"
+    "inquirer": "^11.1.0"
```
Comment (insulineru) [inline package.json:37]: do we really need to bump 2 major versions there? is it still work?

**Review by insulineru — CHANGES_REQUESTED** [review]
Hey, thanks for your work and improvements. 

Comments need to be changed, they are not in English. And also, I am not sure that it is necessary to support large commits. It is bad practice to commit everything in one commit, it is desirable to avoid it. 

It seems that if you analyze each file separately, and then all together, then you will greatly spoil the speed of the solution. I use openai gpt-4o and do not encounter problems with the model not detecting files well, here is an example

image

It would be cool to see visible changes in the speed of prompts (benchmarks) and visual examples of improvements in the fact that analysis through files improves the final result (same files with 2 different versions)

It seems that without this it is not worth merge it in

File: package.json (hunk 5 above)
Comment (Arthur-MONNET) [reply package.json:37]: when I try in list mode, everything works fine 👍

Comment (Arthur-MONNET) [reply package.json:37]: I believe it's crucial to keep dependencies updated to maintain security and stability in our projects, at least that's the way I do it

**Comment by Arthur-MONNET** [comment]
Hello,

I understand your concerns about the speed of generating commit messages and the potential impact on performance. However, I want to reassure you that my proposal should not be directly compared to the current version. Here's why:

### Context of the Split Method

- **Conditional Use**: This file-splitting method is only used if the current method, which relies on a single call to chatGPT based on the diff, exceeds the token limit. Thus, it only intervenes in specific cases where the method cannot be executed in one call.

### Advantages of the Split Method

- **Handling Large Files**: By analyzing each file separately, we avoid having to check automatically generated files that have an astronomical amount of changes that are not important to include in the commit message. This allows us to focus on files that are not too large while excluding the most imposing ones.

- **Autonomy and Simplicity**: This method avoids the need to configure commit-ai with any file exclusion system. The program does it "on its own" by excluding overly large files, simplifying the process and making it more autonomous.

### Why is this Method Useful?

- **For Users Exceeding Token Limits**: This method is particularly useful for those who exceed the token limit when calling GPT. In these cases, the code works as initially intended, with a single call for the complete difference.

- **Backup Solution**: If the initial behavior works well with commit-ai, we continue to use the initial method. This PR is just a backup solution for users who encounter token limit issues.

### Reflection on Commit Practices

It's true that making commits with too many changes is a bad practice. However, this method allows us to cover a broader range of developers and manage various commit situations. If it takes a bit longer, it's often due to users employing less optimal commit methods.

In conclusion, this approach aims to offer additional flexibility without impacting those who don't need it. I am convinced that this solution can enrich our tool and meet the needs of many users.

Thank you for your consideration. 😊
